components/homme/src/theta-l/prim_state_mod.F90

```diff
@@ -310,6 +308,13 @@ subroutine prim_printstate(elem, tl,hybrid,hvcoord,nets,nete)
        global_shared_buf(ie,12)=thetasum_local(ie)
     end do
 
+    if ( .not. theta_hydrostatic_mode ) then
+       call findExtremaWithLevel(elem,wmax_local,'w_i','max',n0,nets,nete)
+       call ParallelMaxWithIndex(wmax_local, hybrid)
+       call findExtremaWithLevel(elem,wmin_local,'w_i','min',n0,nets,nete)
+       call ParallelMinWithIndex(wmin_local, hybrid)
```

Comment:
I like how findExtremaWithLevel uses a string to select min or max. (Might call it Extremum rather than Extrema since it does only one at a time.) Are you _not_ doing that for Paralel*WithIndex to follow parallel_mod conventions? Would be nice to have just one function ParallelExtremumWithLevel.

Reply:
I am not sure about just one function. I don't want to touch the rest of the output for now.